spring-batch-infrastructure/src/test/java/org/springframework/batch/item/xml/StaxEventItemReaderTests.java

```diff
@@ -110,6 +113,41 @@ public void testAfterPropertiesSet() throws Exception {
 		source.afterPropertiesSet();
 	}
 
+	/**
+	 * Regular usage scenario. ItemReader should pass XML fragments to unmarshaller wrapped with StartDocument and
+	 * EndDocument events.
+	 */
+	@Test
+	public void testNoResource() throws Exception {
```

Comment:
There is already a test named `testStrictness` that uses a `NonExistentResource` for the use case related to the original issue. So I think there is no need to add this new test and its related `getBadResource()` method. We can update the existent test to assert that the exception's message now contains the URL of the resource.

I will take care of this on merge.